### PR TITLE
nix::fcntl add FcntlArg::F_READAHEAD for freebsd.

### PR DESCRIPTION
**Pull request by devnexen**
Set/clear the amount of read ahead for the file descriptor.

### CODE REVIEW

File: test/test_fcntl.rs [hunk 1]
```diff
@@ -805,3 +805,22 @@ fn test_f_transferextents() {
...
+#[cfg(target_os = "freebsd")]
+#[test]
+fn test_f_readahead() {
+    use nix::fcntl::*;
+
+    let tmp = NamedTempFile::new().unwrap();
+    // With TMPDIR set with UFS, the vnode name is not entered
+    // into the name cache thus path is always empty.
+    // Therefore, we reopen the tempfile a second time for the test
+    // to pass.
```
Comment (SteveLauC) [inline test/test_fcntl.rs:818]: Why would an empty path fail our test?

File: src/fcntl.rs [hunk 2]
```diff
@@ -825,6 +825,10 @@ pub enum FcntlArg<'a> {
     /// Both descriptors must reference regular files in the same volume.
     #[cfg(apple_targets)]
     F_TRANSFEREXTENTS(RawFd),
+    /// Set or clear the read ahead amount for sequential access or
+    /// disable it with 0 or to system default for any value < 0.
+    #[cfg(target_os = "freebsd")]
+    F_READAHEAD(c_int),
```
Comment (SteveLauC) [inline src/fcntl.rs:831]: Is this something related to the kernel page cache, something used to control the pre-fetch strategy?

File: test/test_fcntl.rs (hunk 1 above)
Comment (devnexen) [reply test/test_fcntl.rs:818]: ah nvm I think this is copy paste from another test..

File: src/fcntl.rs (hunk 2 above)
Comment (devnexen) [reply src/fcntl.rs:831]: it manages file data caching yes. will adjust the comment.

**Review by SteveLauC — APPROVED** [review]

**Comment by SteveLauC** [comment]
Thanks

**Pull request merged**